File: app/src/main/java/io/github/takusan23/kaisendon/APIJSONParse/ActivityPubJSONParse.java
package io.github.takusan23.kaisendon.APIJSONParse;

import org.json.JSONException;
import org.json.JSONObject;

public class ActivityPubJSONParse {

    private String published;
    private String context;

    public ActivityPubJSONParse(String json) {
        json_parse(json);
    }

    public String getContext() {
        return context;
    }

    public String getPublished() {
        return published;
    }

    private void json_parse(String json) {
        try {
            JSONObject jsonObject = new JSONObject(json);
            //tootがあるJSONObject
            JSONObject object = jsonObject.getJSONObject("object");
            context = object.getString("content");
        } catch (JSONException e) {
            e.printStackTrace();
        }
    }

}

File: app/src/main/java/io/github/takusan23/Kaisendon/APIJSONParse/ActivityPubJSONParse.kt
package io.github.takusan23.Kaisendon.APIJSONParse

import org.json.JSONException
import org.json.JSONObject

class ActivityPubJSONParse(json: String) {

    val published: String? = null
    var context: String? = null
        private set

    init {
        json_parse(json)
    }

    private fun json_parse(json: String) {
        try {
            val jsonObject = JSONObject(json)
            //tootがあるJSONObject
            val `object` = jsonObject.getJSONObject("object")
            context = `object`.getString("content")
        } catch (e: JSONException) {
            e.printStackTrace()
        }

    }

}
